mozjs/js/rust/magic_codegen/src/lib.rs

```diff
@@ -310,49 +374,74 @@ lazy_static! {
 fn gen_getter(id: &Option<syn::Ident>,
               ty: &syn::Ty,
               effective_idx: &quote::Ident,
+              js_effective_idx: &quote::Ident,
               inherit: bool,
-              idx: usize) -> quote::Tokens {
-    let getter_str = match *id {
-        Some(ref real_id) => format!("get_{}", real_id.to_string()),
+              idx: usize,
+              name_str: &str) -> (quote::Tokens, quote::Tokens) {
+    let (rust_getter_str, js_getter_str) = match *id {
+        Some(ref real_id) => {
+            let id_str = real_id.to_string();
+            (format!("get_{}", id_str), format!("{}_get_{}", name_str, id_str))
+        },
         None => panic!("Encounter an empty field. Something wrong..."),
     };
-    let getter_name = quote::Ident::from(getter_str.as_str());
-    let getter = if idx == 0 && inherit {
-        quote!{}
+    let rust_getter_name = quote::Ident::from(rust_getter_str.as_str());
+    let js_getter_name = quote::Ident::from(js_getter_str.as_str());
+    let (rust_getter, js_getter) = if idx == 0 && inherit {
+        (quote!{}, quote!{})
     } else {
-        quote! {
-            pub unsafe fn #getter_name (&self, cx: *mut JSContext) -> <#ty as ToFromJsSlots>::Target {
+        let rust_getter = quote! {
+            pub unsafe fn #rust_getter_name (&self, cx: *mut JSContext) -> <#ty as ToFromJsSlots>::Target {
                 let jsobj = self.object;
                 <#ty as ToFromJsSlots>::from_slots(jsobj, cx, #effective_idx)
             }
-        }
+        };
+        let js_getter = quote! {
+            function #js_getter_name() {
+                let res = UnsafeGetReservedSlot(this, #js_effective_idx);
+                return res;
```

Comment:
Ha, nice. Sort of surprised that using the rust quoting would work with JS, but since its at the token level, I guess it makes sense.